server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -294,6 +294,7 @@ public synchronized void onNewCheckpoint(final ReplicationCheckpoint receivedChe
             return;
         }
         updateLatestReceivedCheckpoint(receivedCheckpoint, replicaShard);
```

Comment:
we should remove `latestReceivedCheckpoint` map from this class and rely only on the one in segmentReplicator now.